hibernate-core/src/test/java/org/hibernate/test/sql/function/JpaFunctionTest.java

```diff
@@ -80,7 +81,7 @@ public void testWithoutCommaFail() {
 			} );
 		}
 		catch ( Exception e ) {
-			assertEquals( QuerySyntaxException.class, e.getCause().getClass() );
+			assertEquals( QueryException.class, e.getCause().getClass() );
```

Comment:
in `o.h.hql.internal.ast. parse(boolean filter)` I would add
```
if ( hqlAst == null ) {
   parser.getParseErrorHandler().throwQueryException();
} 
```
after 
`final AST hqlAst = parser.getAST();`
this should avoid the change from `QuerySyntaxException` and `QueryException` and probably a better error cause message.

With this change the ASTParserLoadingTest# testOrderByExtraParenthesis () catch clause will become
	```
catch (IllegalArgumentException e) {
			final Throwable cause = e.getCause();
			assertThat( cause, instanceOf( QuerySyntaxException.class ) );
			assertTrue( cause.getMessage().contains( "expecting EOF, found ')'" ) );
		}
```